bl_mcu_sdk_bl808/examples/low_load/main.c

```diff
@@ -172,25 +173,19 @@ static void lp_ipc_handler(uint32_t src)
     MSG("%s: src: 0x%08x\r\n", __func__, src);
 }
 
+static uint32_t ipc_irqs[32] = {
+    [BFLB_IPC_DEVICE_SDHCI] = SDH_IRQn,
+    [BFLB_IPC_DEVICE_UART2] = UART2_IRQn,
+    0,
+};
+
 static void d0_ipc_handler(uint32_t src)
 {
-    int bit;
-
-    for (bit = 0; bit < 32; bit++)
+    for (int bit = 0; src != 0 && bit < 32; bit++)
     {
-        if ((src >> bit) & 1)
-        {
-            switch (bit)
-            {
-                case 0:
-                    CPU_Interrupt_Enable(SDH_IRQn);
-                    break;
-
-                case 1:
-                    CPU_Interrupt_Enable(UART2_IRQn);
-                    break;
-            }
-        }
+        if ((src >> bit) & 1) CPU_Interrupt_Enable(ipc_irqs[bit]);
```

Comment:
Add a check to make sure ipc_irqs[bit] is not 0 before calling CPU_Interrupt_Enable() to handle the case where we get an EOI for a non existing device.

Also, change the IRQHandlers above to also use the device id's from dt-bindings like this:
IPC_M0_Trigger_D0(1 <<  BFLB_IPC_DEVICE_XXX);

Reply:
Completed